tools/precommit.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Samsung Electronics Co., Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import sys
+import os
+from common_py import path
+from common_py.system.filesystem import FileSystem as fs
+from common_py.system.executor import Executor as ex
+from common_py.system.platform import Platform
+
+TESTS=['host', 'rpi2', 'nuttx', 'darwin', 'misc']
+BUILDTYPES=['debug', 'release']
+
+
+def parse_option():
+    parser = argparse.ArgumentParser(
+         description='IoT.js pre-commit script.',
+         epilog='If no arguments are given, runs full test.')
+    parser.add_argument('--test', choices=TESTS, action='append')
+    parser.add_argument('--buildtype', choices=BUILDTYPES, action='append')
+
+    option = parser.parse_args(sys.argv[1:])
+    if option.test is None:
+        option.test = TESTS
+    if option.buildtype is None:
+        option.buildtype = BUILDTYPES
+    return option
+
+
+def setup_nuttx_root(nuttx_root):
+    # Step 1
+    fs.maybe_make_directory(nuttx_root)
+    fs.chdir(nuttx_root)
+    if fs.exists('nuttx'):
+        fs.chdir('nuttx')
+        ex.check_run_cmd('git', ['pull'])
+        fs.chdir('..')
```

Comment:
Since our nuttx build guide doesnt' specify the exact version of nuttx, I think it would be better to make the buildbot to follow the latest version of nuttx.
When buildbot fails, that would be the right time to fix the nuttx version or update our nuttx build script.